client_tipc.c

```diff
@@ -1067,6 +1073,119 @@ static struct storage_client_session *chan_context_to_client_session(struct ipc_
 	return session;
 }
 
+static rpmb_block_t *get_rpmb_block(void)
+{
+	int ret;
+	rpmb_block_t *rpmb_blk;
+
+	rpmb_blk = (rpmb_block_t *)malloc(RPMB_BLOCK_SIZE);
+	if (!rpmb_blk) {
+		SS_ERR("%s: rpmb_blk malloc fail!\n", __func__);
+		return NULL;
+	}
+
+	ret = rpmb_read(g_rpmb_state, (void *)rpmb_blk, RPMB_BLOCK_INFO_INDEX, 1);
+	if (ret) {
+		SS_ERR("%s: read rpmb block table fail!\n", __func__);
+		free(rpmb_blk);
+		return NULL;
+	}
+
+	if (strncmp((const char *)rpmb_blk->signature, "BARA", 4)) {
+		SS_ERR("%s: Invalid signature!\n", __func__);
+		free(rpmb_blk);
+		return NULL;
+	}
+
+	if (!(rpmb_blk->flag & ATTKB_PRESENCE)) {
+		SS_ERR("%s: Attkb is not presence!\n", __func__);
+		free(rpmb_blk);
+		return NULL;
+	}
+
+	if (rpmb_blk->attkb_size == 0 || rpmb_blk->attkb_size > ATTKB_MAX_SIZE) {
```

Comment:
i'm wondering if there is any hard/strong reason that limit 16KB in our coding?

Reply:
checked CSE source code, it's limited to 15.5KB.
For Celadon, fastboot will also have a size check, to prevent overwriting the latter file system RPMB.